src/googleError.ts

```diff
@@ -25,6 +25,32 @@ export class GoogleError extends Error {
   statusDetails?: string | protobuf.Message<{}>[];
   reason?: string;
   domain?: string;
+  errorInfoMetadata?: {string: string};
+
+  // Parse details field in google.rpc.status wire over gRPC medatadata.
+  // Promote google.rpc.ErrorInfo if exist.
+  static parseGRPCStatusDetails(err: GoogleError): GoogleError {
+    const decoder = new GoogleErrorDecoder();
+    try {
+      if (err.metadata && err.metadata.get('grpc-status-details-bin')) {
+        const statusDetailsObj: GRPCStatusDetailsObject =
+          decoder.decodeGRPCStatusDetails(
+            err.metadata.get('grpc-status-details-bin') as []
+          );
+        if (statusDetailsObj && statusDetailsObj.details.length > 0) {
+          err.statusDetails = statusDetailsObj.details;
+        }
+        if (statusDetailsObj && statusDetailsObj.errorInfo) {
+          err.reason = statusDetailsObj.errorInfo.reason;
+          err.domain = statusDetailsObj.errorInfo.domain;
+          err.errorInfoMetadata = statusDetailsObj.errorInfo.metadata;
+        }
+      }
+    } catch (decodeErr) {
+      return err;
+    }
+    return err;
```

Comment:
nit: since you `return err` both in `try` and in `catch`, you can just have an empty `catch` (with a comment `// ignoring the error`) and a single `return err` in the end of the function.

Reply:
SG